pkg/util/http/client.go

```diff
@@ -1,23 +1,73 @@
 package http
 
 import (
-	nethttp "net/http"
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
 	"net/url"
 	"path"
+
+	"github.com/pkg/errors"
+
+	error_types "github.com/kumahq/kuma/pkg/core/rest/errors/types"
 )
 
 type Client interface {
-	Do(req *nethttp.Request) (*nethttp.Response, error)
+	Do(req *http.Request) (*http.Response, error)
 }
 
-type ClientFunc func(req *nethttp.Request) (*nethttp.Response, error)
+type ClientFunc func(req *http.Request) (*http.Response, error)
 
-func (f ClientFunc) Do(req *nethttp.Request) (*nethttp.Response, error) {
+func (f ClientFunc) Do(req *http.Request) (*http.Response, error) {
 	return f(req)
 }
 
+type TokenClient struct {
+	client Client
+	url    string
+}
```

Comment:
IMHO this is too specific to have this in `pkg/util/http`.
I consider `pkg/util` a package that has a code that the standard library/libaries that we depend lacks in functionality.

How about we move this to `pkg/tokens/client.go` ?

Reply:
Nice, it done :) !